packages/web/src/scss/tools/_form-fields.scss

```diff
@@ -0,0 +1,136 @@
+
+    @if $fieldType == 'inline' {
+        @include typography.generate(form-fields-theme.$input-typography);
+
+        display: inline-block;
+    }
+
+    @if $fieldType == 'box' {
+        @include typography.generate(form-fields-theme.$box-field-label-typography);
+
+        display: block;
+        margin-bottom: form-fields-theme.$input-spacing;
+    }
+}
+
+@mixin label-disabled() {
+    color: form-fields-theme.$label-color-disabled;
+}
+
+@mixin label-hidden() {
+    @include accessibility.hide-text();
+}
+
+@mixin label-required() {
+    content: '*';
+    color: form-fields-theme.$label-required-color;
+}
+
+@mixin input($fieldType) {
+    @if $fieldType == 'inline' {
+        appearance: none;
+        margin: form-fields-theme.$inline-field-input-margin;
+        color: form-fields-theme.$inline-field-input-color-unchecked;
+        border: form-fields-theme.$inline-field-input-color-default solid currentcolor;
+
+        &:focus-visible {
+            outline: 0;
+            box-shadow: form-fields-theme.$input-focus-shadow;
+        }
+    }
+
+    @if $fieldType == 'box' {
+        @include typography.generate(form-fields-theme.$input-typography);
+
+        display: block;
+        width: 100%;
+        padding: form-fields-theme.$box-field-input-padding-y form-fields-theme.$box-field-input-padding-x;
+        color: form-fields-theme.$box-field-input-color-default;
+        border: form-fields-theme.$box-field-input-border-width
+            form-fields-theme.$box-field-input-border-style
+            form-fields-theme.$box-field-input-border-color-default;
+        border-radius: form-fields-theme.$input-border-radius;
+        background: form-fields-theme.$box-field-input-background;
+
+        &::placeholder {
+            color: form-fields-theme.$box-field-input-placeholder-color-default;
+            opacity: 1;
+        }
+    }
+}
+
+@mixin input-disabled() {
+    color: form-fields-theme.$input-color-disabled;
+}
+
+@mixin message() {
+    @include typography.generate(form-fields-theme.$message-typography);
+
+    display: block;
+    margin-top: form-fields-theme.$message-margin-top;
+    color: form-fields-theme.$message-color-default;
+}
+
+@mixin message-disabled() {
+    color: form-fields-theme.$message-color-disabled;
+}
+
+@mixin box-field-variants($fieldName) {
+    @each $variant-name, $variant-value in form-fields-settings.$box-field-variants {
+        :is(.#{$fieldName}--#{$variant-name}, .#{$fieldName}.has-#{$variant-name}) > .#{$fieldName}__input {
+            border-color: map.get($variant-value, input-border-color);
+        }
+
+        @if $fieldName == 'TextField' {
+            :is(.#{$fieldName}--#{$variant-name}, .#{$fieldName}.has-#{$variant-name})
+                .#{$fieldName}__passwordToggle
+                > .#{$fieldName}__input
+                ~ .#{$fieldName}__passwordToggle__button::before {
+                border-color: map.get($variant-value, input-border-color);
```

Comment:
Please refactor the theming of variants to use custom properties instead of generating these wild selectors for each variant. This mixin would then only assign colors to these custom properties.

Reply:
We've agreed create an [issue](https://jira.lmc.cz/browse/DS-478) for this comment.